esptool.py

```diff
@@ -2392,7 +2392,7 @@ def version(args):
 #
 
 
-def main(custom_commandline=None):
+def main(custom_commandline):
```

Comment:
I've seen there is existing code out there which writes `sys.argv` and then calls `esptool.main()` (this behaviour predates the `custom_commandline` argument).

To avoid breaking this code, I'd prefer if the default argument was kept and somewhere in main() it does:

```
if custom_commandline is None:
   custom_commandline = sys.argv[1:]
```

(`sys.argv[1:]` can't be the default argument because it needs to be expanded when the function is called, not when esptool is imported).

If you're feeling super generous, it may be a good improvement to rename `custom_commandline` to something like `argv`, if it's going to be passed around in multiple places.